security/kubernetes-nodes.md

```diff
@@ -0,0 +1,280 @@
+For example, if we want to add the label **environment=dev** to nodes named node1 and node2:
+
+```bash
+kubectl label node node1 environment=dev
+kubectl label node node2 environment=dev
+```
+
+With the labels in place and automatic host endpoints enabled, host endpoints for node1 and node2 will be updated with the **environment=dev** label.
+We can write policy to select that set of nodes with a combination of selectors:
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: some-nodes-policy
+spec:
+  selector: has(kubernetes-host) && environment == 'dev'
+  <rest of the policy>
+```
+
+### Tutorial
+
+This tutorial will lock down Kubernetes node ingress to only allow SSH and required ports for Kubernetes to function.
+We will apply two policies: one for the master nodes. and one for the worker nodes.
+
+> Note: Run this tutorial only on a sandbox cluster; using it on a real cluster can disrupt traffic.
+{: .alert .alert-danger }
+
+> Note: This tutorial was tested on a cluster created with kubeadm v1.18.2 on AWS, using a "stacked etcd" [topology](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/ha-topology/). Stacked etcd topology means the etcd pods are running on the masters. kubeadm uses stacked etcd by default.
+>
+> If your Kubernetes cluster is on a different platform, is running a variant of Kubernetes, or is running a topology with an external etcd cluster,
+> please review the required ports for master and worker nodes in your cluster and adjust the policies in this tutorial as needed.
+{: .alert .alert-info }
+
+First, let's restrict ingress traffic to the master nodes. The ingress policy below contains three rules.
+The first rule allows access to the API server port from anywhere. The second rule allows access to the Kubernetes
+control plane from localhost. These control plane processes includes the etcd server client API, the scheduler, and the controller-manager. This rule
+also whitelists localhost access to the kubelet API and calico/node health checks. And the final rule allows the etcd pods
+to peer with each other.
+
+If you have not modified the failsafe ports, you should still have SSH access to the nodes after applying this policy.
+Now apply the ingress policy for the Kubernetes masters:
+
+```
+calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: ingress-k8s-masters
+spec:
+  selector: has(node-role.kubernetes.io/master)
+  # This rule allows ingress to the Kubernetes API server.
+  ingress:
+  - action: Allow
+    protocol: TCP
+    destination:
+      ports:
+      # kube API server
+      - 6443
+  # This rule allows traffic to Kubernetes control plane ports
+  # from localhost. The health check port for calico/node is also whitelisted
+  - action: Allow
+    protocol: TCP
+    destination:
```

Comment:
Description says "from local host" but the policy seems to be "to local host"?

Reply:
29a529de33528525254970a65403235eb380f5ca